designer/server/src/test/scala/pl/touk/nussknacker/ui/api/helpers/EspItTest.scala

```diff
@@ -330,7 +332,14 @@ trait EspItTest extends LazyLogging with WithHsqlDbTesting with TestPermissions
 
   private def saveAndGetId(process: CanonicalProcess, category: String, isSubprocess: Boolean, processingType: ProcessingType = Streaming): Future[ProcessId] = {
     val processName = ProcessName(process.id)
-    val action = CreateProcessAction(processName, category, process, processingType, isSubprocess, None)
+    val action = CreateProcessAction(
+      processName,
+      category,
+      process,
+      processingType,
+      isSubprocess,
+      ComponentsUsageHelper.computeUsagesForScenario(process),
```

Comment:
1. WDYT about moving this `computeUsagesForScenario` into repository level? Usages are just an index, if we have json indexes in DB, we won't need it
a) If you don't like this concept, at least extract `Create/UpdateProcessAction.apply` factory methods computing usages based on process - it will be haderd to desynchronize it
2. WDYT about moving `computeUsagesForScenario` logic into `CanonicalProcess` and having just a `CanonicalProcess.componentUsages`?

Reply:
Good point! I chose option 1 because moving the logic into `CanonicalProcess` would unnecessarily expose component usages in the public API.